### DISCUSSION TITLE
Stable Forge Support?

### DISCUSSION THREAD

**Discussion by Gecktendo** [Kilvoctu/aiyabot#272] [Ideas]
I know that Vladomatic and Auto1111 are supported, but is there any way we can get Forge support as well?

**Comment by wizz13150** [comment]
Yoh,

I'm running on Forge for a while now. Still using the bot daily.
But we need to be ready for Kontext.
Just uploaded my messy code.
Made a shitload of changes. Don't know where to start . The Forge specific commits are totally burried in the mass..

**_The repo is probably not "ready-to-run" with a git clone._** Many hardcoded values too.
Added some main features like a chatbot (able to run on AMD, able to trigger a generation and use options like hires, ratios or adetailer. I'll refactor the chatbot soon for a better model and vision), trained better models for /Generate, a Details++ option for decent upscales/refines using USDU and Soft Inpainting. 
An infinite generation option for /Draw (very cool), a random prompt option, a random style option (can cumulate these 3 options as well), size ratios presets, distilled CFG, schedulers, a prompt prefix option (e.g. LoRas triiger words or broken models. :cough: Pony :cough:), an adaptive color correction and sharpening for Details++ [...]

It's a mess but the code is there. :)

To run Forge imply to edit settings.py and stablecog.py. That's it.
A better/full implementation is a 'lil more complex.

Here is my fork :
https://github.com/wizz13150/aiyabot
Gallery 100% from the bot, 99% from ZavyChromaXL :
https://civitai.com/user/_Wizz_/images

Image from the bot and ZavyChromaXL (SDXL) using Details++ (1100sec on a RTX3090) : 

![76112739](https://github.com/user-attachments/assets/f3bfa1b6-dac0-4144-8e2e-f12c1f8c410a)
(I had to resize the pic down to 70% to post it here. Open in a new tab and zoom-in.)

Cheers ! 🥂
